android/titanium/src/java/org/appcelerator/titanium/TiLaunchActivity.java

```diff
@@ -96,7 +97,11 @@ protected void loadScript()
 	{
 		try {
 			String fullUrl = resolveUrl(this.url);
-			KrollRuntime.getInstance().runModule(KrollAssetHelper.readAsset(fullUrl), fullUrl, activityProxy);
+			if (KrollAssetHelper.assetExists(fullUrl)) {
+				KrollRuntime.getInstance().runModule(KrollAssetHelper.readAsset(fullUrl), fullUrl, activityProxy);
+			} else {
+				KrollRuntime.getInstance().runModule("global._startup(global)", fullUrl, activityProxy);
```

Comment:
Don't you need to check against the specific url you want to hijack with a snapshot? Or are we just assuming that ti.main.js won't actually exist because we're not copying it during app build?

Reply:
I think it's okay for the `else` block to assume the snapshot exists. The old code was already assuming "ti.main.js" was being copied-in by our CLI anyways. And this is the route our mocha test suite will take, so, we'll know when something goes wrong.

Also, "fullUrl" will exist if we copy the "titanium_mobile/common/Resources" folder ("ti.main.js" and it's "ti.internal" directory) to the app project (to be ran via `appc run`) for easy testing/debugging of our core JS changes. This is something I want to make sure we continue to support for "productivity" reasons. In this case the "ti.main.js" we drop in will be favored and the snapshot will be ignored.